File: fontes/cliente/dotNet/Thewar/br/com/thewar/lang/ASubject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace br.com.thewar.lang
{
    /// <summary>
    /// 
    /// </summary>
    public abstract class ASubject
    {
        #region Métodos
        /// <summary>
        /// 
        /// </summary>
        /// <param name="observer"></param>
        public void Attach(IObserver observer)
        {
            listObservers.Add(observer);
        }
        /// <summary>
        /// 
        /// </summary>
        /// <param name="observer"></param>
        public void Detach(IObserver observer)
        {
            listObservers.Remove(observer);
        }
        /// <summary>
        /// 
        /// </summary>
        public void Notify()
        {
            foreach (IObserver o in listObservers)
            {
                o.Update();
            }
        }
        #endregion

        #region Atributos
        /// <summary>
        /// 
        /// </summary>
        private List<IObserver> listObservers = new List<IObserver>();
        #endregion

        #region Propriedades
        /// <summary>
        /// 
        /// </summary>
        public string SubjectState { get; set; }
        #endregion
    }
}

File: fontes/servidor/java/server/src/main/java/br/com/thewar/server/lang/ASubject.java
package br.com.thewar.server.lang;

import java.util.ArrayList;
import java.util.List;

public abstract class ASubject {
	
	//
	private List<IObserver> list = new ArrayList<IObserver>();
	//
	private String subjectState;

	/**
	 * 
	 * @param observer
	 */
	public void Attach(IObserver observer) {
		list.add(observer);
	}

	/**
	 * 
	 * @param observer
	 */
	public void Detach(IObserver observer) {
		list.remove(observer);
	}

	/**
	 * 
	 */
	public void Notify() {
		for (IObserver ob : list) {
			ob.Update();
		}
	}
	
	/**
	 * 
	 * @param subjectState
	 */
	public void Notify(String subjectState) {
		this.subjectState = subjectState;
		
		for (IObserver ob : list) {
			ob.Update();
		}
	}
	
	public String getSubjectState() {
		return subjectState;
	}
	
	public void setSubjectState(String subjectState) {
		this.subjectState = subjectState;
	}
}
